File: DSA_Questions/dsa100/twoSum.js
const twoSum = (nums, target) => {
	let indexOf = {};
	// track the indexes

	for (let i = 0; i < nums.length; i++) {
		let complement = target - nums[i];

		if (complement in indexOf) {
			return [indexOf[complement], i];
		}

		indexOf[nums[i]] = i;
	}
};

let nums = [2, 7, 11, 15];
let target = 9;

console.log(twoSum(nums, target));

File: DSA_Questions/dsa100/twoSum.rb
nums = [2,7,11,15]
target = 26


def two_sum(nums, target)
  indexOf = {}
  nums.each_with_index do |n, i|
    if indexOf[target - n]
      return indexOf[target - n], i
    end
    indexOf[n] = i
  end
end
p two_sum(nums, target)

# colors = ['red', 'green', 'blue']
# colors.each_with_index do |item, index|
# 	p "#{index}:#{item}" 
# end